docs/Interoperable-RA-TLS-cert-edvidence-formats.md

```diff
@@ -4,17 +4,19 @@ Interoperable RA-TLS X.509 Cert and Evidence Formats
 # X.509 Cert with Evidence and Endorsements Extensions
 
 A X.509 cert for RA-TLS can either be self-signed or CA-signed. It must contain an extension for evidence, and can (optionally) contain an extension for endorsements. The evidence and endorsement extensions are defined in the latest [TCG DICE specification draft](https://members.trustedcomputinggroup.org/wg/DICE/document/36858).
+
 - Evidence (required): X.509 cert extension, a byte string of encoded tagged CBOR data
-    - OID: `tcg-dice-tagged-evidence` (2.23.133.5.4.9)
+    - OID: `tcg-dice-conceptual-message-wrapper` (2.23.133.5.4.9)
     - CBOR tag: a CBOR tag registered with IANA or some other registry serves as evidence format ID
-    - CBOR data: this data covered by the tag contains the evidence (including custom claims). Its format is defined by the CBOR tag. 
+    - CBOR data: this data covered by the tag contains the evidence (including custom claims). Its format is defined by the CBOR tag.
+    - Note: the DICE specification states that the criticality flag of this extension SHOULD be marked critical, but popular TLS libraries (such as openssl and mbedtls) will fail in verification of an X.509 cert with this extension if its criticality flag is set. To overcome this limitation, the criticality flag will be cleared.
```

Comment:
@dimakuv @KB5201314 the embedtls page describes a mitigation by using `mbedtls_x509_crt_parse_der_with_ext_cb()`. The openssl page also mentions a `-ignore_critical` option for `openssl-verify` utility, indicating that there is a way handle unsupported critical extensions. Have you guys looked into these options. The best is to be able to set the cert extension for quote to be, as described in the DICE spec.

Reply:
> Have you guys looked into these options. 

For OpenSSL, if a certificate contains a critical extension that OpenSSL doesn't recognize, it will report an `X509_V_ERR_UNHANDLED_CRITICAL_EXTENSION` error, see [[1]](https://github.com/openssl/openssl/blob/a63fa5f711f1f97e623348656b42717d6904ee3e/crypto/x509/x509_vfy.c#L489-L491) and [[2]](https://github.com/openssl/openssl/blob/a63fa5f711f1f97e623348656b42717d6904ee3e/crypto/x509/v3_purp.c#L596-L599). This error can be ignored in verify_callback, just like `-ignore_critical` option do. However, there is no way to determine which extension caused this error. If we choose to ignore all of them, we may miss other unrecognized critical extensions, which I think would be a security risk.

Reply:
Based on my experiments with mbedTLS, we can't set the criticality flag. See #9 (comment)

Reply:
Let's have this criticality flag cleared, until the TLS library support issue has been resolved.